### ISSUE TITLE
SSL Certs for nodejs.org and iojs.org

### ISSUE THREAD

**Issue by rvagg** [nodejs/build#812]
<img width="716" alt="screen shot 2017-07-24 at 12 28 44 am" src="https://user-images.githubusercontent.com/495647/28512530-234b3164-7007-11e7-8481-46aa75b71a72.png">

Sorry to just dump this raw here but I have no time this week to even think about this so I'm hoping someone else can take it on. I'm getting renewal notices for nodejs.org and iojs.org certs but I don't know if we actually use them still or not. If we do, we need to decide whether to renew (they are not expensive through gogetssl and the Foundation can pay) or go with an alternative. I _think_ we may be able to use CloudFlare to issue certs on demand for us. Or, as @jbergstroem has suggested we got for letsencrypt, he advocated for using https://github.com/lukas2511/dehydrated for that job but neither of us have had the time to do anything about it.

In other words, can someone investigate and propose a path forward, pretty please?

**Comment by ljharb** [comment]
just to underscore the urgency, if the SSL certs for either of these sites stop working, every single node.js travis-ci test will instantly begin to fail. yay!

**Comment by rvagg** [comment]
How pessimistic of you @ljharb! We won't let it fail. This is merely an early call for help, if nothing moves then I'll just find some spare hours to do it myself and that might just mean paying for replacement certs cause it's the quickest solution.

**Comment by ljharb** [comment]
Oh I trust yall will figure it out in plenty of time :-)

**Comment by jevtovich** [comment]
What do you think about [Let's Encrypt SSL](https://letsencrypt.org/)? I think they are free, and the foundation can donate some if they want to.

**Comment by MylesBorins** [comment]
I'd say we should top up the infra we have that already works and start the process of moving over to lets encrypt

**Comment by rvagg** [comment]
I think I agree with that sentiment, it's not expensive to renew these multi-year ones (I'll get the actual price for you soon) and then we can tinker with letsencrypt. Unfortunately the need to regularly renew letsencrypt certs makes it a non-trivial exercise and with all of the things on our plate the moment we should probably give ourselves some room to move!

**Comment by MylesBorins** [comment]
@rvagg I have setup a cron job on one of my own servers to keep lets-encrypt up to date... although I will be honest that I think a hosted service is a better solution... would rather have an SLA and not be responsible for taking Travis offline. At the same time I really want to support let's encrypt... so I'm unsure of the best path forward

**Comment by rvagg** [comment]
letsencrypt is a Linux Foundation project, a path forward might be to reach into LF connections and find some help to making a successful collaboration, even if that means having them hold our hands on best practices to make sure we can reach the kind of reliability we need without having to have us constantly worrying about it!

<img width="742" alt="screen shot 2017-08-02 at 4 29 52 pm" src="https://user-images.githubusercontent.com/495647/28860418-e2644ed8-779f-11e7-96bd-6c2734cce771.png">

Unfortunately it's not as cheap as I remember but this is for wildcard certs for 24 months. It's $70.95 if we go for 12 months.

@MylesBorins @mhdawson @jasnell @nodejs/build we should probably get TSC agreement on this kind of expenditure right? How about we go for 12 month certs x 2 = $141.90 total, with the plan to replace it with a solid letsencrypt setup during that timeframe. What think you?

**Comment by MylesBorins** [comment]
LGTM

**Comment by piccoloaiutante** [comment]
Sounds good to me

**Comment by mhdawson** [comment]
LGTM on renewing now and then figuring out a migration path.

**Comment by mhdawson** [comment]
Opened https://github.com/nodejs/TSC/issues/302 for the formal TSC approval side of things.  

@rvagg what is the standard way that we have done this in the past and then made sure that we reimbursed whoever does the payment ?

**Comment by jasnell** [comment]
I have asked the foundation folks if funds are available or if an allocation is required. I agree with the short term approach here for the next year, but would definitely like to see us move to the letsencrypt option in the future.

**Comment by rvagg** [comment]
Hey, so digging in further to our GoGetSSL account I've found that we have a paid 24-month wildcard cert that was never issued! This was intended for `*.libuv.org` which is still cert-less and we should fix along with this current round of updates.

So, how's this for a plan:
 * Use the 24-month wildcard we currently own for a renewal on `*.nodejs.org`
 * Use new NF funds for 2 x 12-month wildcards for `*.iojs.org` and `*.libuv.org`
 * Over the next 12 months migrate iojs and libuv over to a letsencrypt setup
 * Give ourselves a window of up to 12 months from then to assess stability and operational costs of the letsencrypt setup before migrating nodejs.

/cc @saghul @cjihrig

**Comment by jbergstroem** [comment]
I will have more time come mid-september and can do the letsencrypt implementation then. Will opt for DNS validation to simplify spreading the certificates.

**Comment by rvagg** [comment]
I've gone ahead and ordered the two additional certificates and requested certificates for the 3 wildcard domains.

**Comment by rvagg** [comment]
Certificates have been installed. nodejs.org (server and cloudflare so it's active for all subdomains like benchmarking.nodejs.org), iojs.org and dist.libuv.org which now has strict https which is a deviation because it was only http until now. Will talk to the libuv folks about bringing in libuv.org from gh-pages onto the server so we can https it too but that's not as urgent I think.

We have a year for *.iojs.org and *.libuv.org and 2 years for *.nodejs.org. So we need to start planning for a migration to new certificates.

I paid for the certs and have submitted an expense claim to the LF for NF funds for 2 x 70.95 USD.

Would appreciate having other folks testing my work and giving a :+1: that all is good.

**Comment by refack** [comment]
`nodejs.org` / `benchmarking.nodejs.org` (arbitrary `benchmarks.nodejs.org` redirects to main site 👍 )
![image](https://user-images.githubusercontent.com/96947/29316526-557668e8-8196-11e7-8cfc-61feffe1c88a.png)
`iojs.org`
![image](https://user-images.githubusercontent.com/96947/29316659-d51b3902-8196-11e7-829d-c1814a1ad40b.png)
`https://dist.libuv.org`
![image](https://user-images.githubusercontent.com/96947/29316725-069e2e44-8197-11e7-8ee1-77be1b03da4d.png)

**Comment by Spetnik** [comment]
Just FYI: AWS offers free TLS certs (SNI-only) when using CloudFront or ELB to serve your application. Obviously if you're not already using AWS it opens up a whole bunch of other issues, but I figured I throw it out there.
